framework/src/outputs/Checkpoint.C

```diff
@@ -72,6 +80,52 @@ Checkpoint::directory() const
   return _file_base + "_" + _suffix;
 }
 
+void
+Checkpoint::outputStep(const ExecFlagType & type)
+{
+  // Output is not allowed
+  if (!_allow_output && type != EXEC_FORCED)
+    return;
+
+  // If recovering disable output of initial condition, it was already output
+  if (type == EXEC_INITIAL && _app.isRecovering())
+    return;
+
+  // Check whether we should output, then do it.
+  if (shouldOutput(type))
+  {
+    TIME_SECTION("outputStep", 2, "Outputting Step");
+    output(type);
+  }
+}
+
+bool
+Checkpoint::shouldOutput(const ExecFlagType & type)
+{
+  // Check if the checkpoint should "normally" output, i.e. if it was created
+  // through checkpoint=true
+  bool shouldOutput = (onInterval() || type == EXEC_FINAL) ? FileOutput::shouldOutput(type) : false;
+
+  // Lambda helper to check the autosave flag
+  // If this is a pure system-created autosave through AutoCheckpointAction,
+  // then sync across processes and only output one time per signal received.
+  auto checkInterrupt = [&]()
```

Comment:
Glad you are learning - let's remove unless you need this to be called from multiple places, or really need the lambda part of this (the closure part). Right now it's parameterless and isn't saving any kind of closure so it's superfluous.